File: backend/app/schemas/task_plan.py
from pydantic import BaseModel, Field


class TaskPlan(BaseModel):
    topic: str
    industry: str | None = None
    target_product: str | None = None
    competitors: list[str] = Field(default_factory=list)
    analysis_dimensions: list[str] = Field(default_factory=list)
    report_depth: str = "standard"
    output_language: str = "zh-CN"
    auto_discover_competitors: bool = False
    data_sources: list[str] = Field(default_factory=list)


class TaskPlanParseRequest(BaseModel):
    user_input: str


class TaskPlanParseResponse(BaseModel):
    task_plan: TaskPlan

File: frontend/src/types/taskPlan.ts
export interface TaskPlan {
  topic: string
  industry?: string | null
  target_product?: string | null
  competitors: string[]
  analysis_dimensions: string[]
  report_depth: 'simple' | 'standard' | 'deep' | string
  output_language: string
  auto_discover_competitors: boolean
  data_sources: string[]
}
